compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2514,6 +2514,12 @@ fn add_link_args(link_args: &mut LinkArgs, flavor: LinkerFlavor, args: &[&'stati
     add_link_args_iter(link_args, flavor, args.iter().copied().map(Cow::Borrowed))
 }
 
+impl TargetOptions {
+    pub fn supports_comdat(&self) -> bool {
+        !self.is_like_aix && !self.is_like_osx
+    }
```

Comment:
Some amount of YOLO is probably unavoidable here, but explicitly mentioning the object file formats should make life easier for anyone who has to think about this code in the future.

```suggestion
    pub fn supports_comdat(&self) -> bool {
        // XCOFF and MachO don't support COMDAT.
        !self.is_like_aix && !self.is_like_osx
    }
```